pkg/syncer/metadatasyncer.go

```diff
@@ -831,6 +842,345 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
+				Namespace: sqPeriodicSync.Namespace,
+				Name:      sqPeriodicSync.Name},
+				sqPeriodicSync)
+			if err != nil {
+				log.Errorf("failed to fetch StorageQuotaPeriodicSync instance with name %q Error: %+v",
+					StorageQuotaPeriodicSyncInstanceName, err)
+				return
+			}
+		} else {
+			log.Errorf("failed to fetch StorageQuotaPeriodicSync instance with name %q Error: %+v",
+				StorageQuotaPeriodicSyncInstanceName, err)
+			return
+		}
+	}
+	// start cron job
+	s := gocron.NewScheduler(time.UTC)
+	job, err := s.Every(sqPeriodicSync.Spec.SyncIntervalInMinutes).Minute().Do(func() {
+		go syncStorageQuotaReserved(ctx, cnsOperatorClient, metadataSyncer)
+	})
+	if err != nil {
+		log.Errorf("cannot start periodic sync for storage quota instance %q, Error: %+v ", sqPeriodicSync.Name, err)
+		return
+	}
+	// Make sure a single job runs(no overlapping jobs)
+	job.SingletonMode()
+	s.StartAsync()
+}
+
+func syncStorageQuotaReserved(ctx context.Context,
+	cnsOperatorClient client.Client, metadataSyncer *metadataSyncInformer) {
+	log := logger.GetLogger(ctx)
+	lastSyncTime := metav1.NewTime(time.Now())
+	// fetching storagepolicyquota
+	spqList := &storagepolicyv1alpha2.StoragePolicyQuotaList{}
+	err := cnsOperatorClient.List(ctx, spqList, &client.ListOptions{})
+	if err != nil {
+		log.Errorf("failed to fetch storage policy quota instances, Error: %+v", err)
+		return
+	}
+	log.Infof("Retrieved StoragePolicyQuota instances for syncing", "Count: %d", len(spqList.Items))
+	namespaces := make(map[string]string)
+	// create namespace map to be able to calculate reserved namespace wise
+	for _, spq := range spqList.Items {
+		namespaces[spq.Namespace] = ""
+	}
+	// fetch pv list
+	volumeMap, err := fetchPVs(ctx, metadataSyncer)
+	if err != nil {
+		log.Errorf("failed to fetch persistentvolume instances, Error: %+v", err)
+		return
+	}
+	totalStoragePolicyReserved := make(map[string]*resource.Quantity)
+	expectedReservedValues := []sqperiodicsyncv1alpha1.ExpectedReservedValues{}
+	// loop over namespaces and calcuate reserved values
+	for ns := range namespaces {
+		log.Infof("processing storage quota sync for namespace %q", ns)
+		// calculate storagepolicyusage reserved values for given namespace
+		spuVmServiceReserved, err := calculateVMServiceStoragePolicyUsageReservedForNamespace(ctx, cnsOperatorClient, ns)
+		if err != nil {
+			log.Errorf("error while calculating expected VmService StoragePolicyUsage reserved value, Error: %v", err)
+			return
+		} else if spuVmServiceReserved != nil {
+			totalStoragePolicyReserved = mergeStoragePolicyReserved(spuVmServiceReserved, totalStoragePolicyReserved)
+		}
+		// calculate pending and under expansion PVC's reserved values for given namespace
+		pvcReserved, err := calculatePVCReservedForNamespace(ctx, cnsOperatorClient, volumeMap, ns, metadataSyncer)
+		if err != nil {
+			log.Errorf("error occured while calculating expected PVC reserved value  for given namespace %q, "+
+				"Error: %v", ns, err)
```

Comment:
Add continue here and for snapshot calculations as well, so that it will continue for next namespace